service_configuration_lib/spark_config.py

```diff
@@ -252,22 +249,17 @@ def _append_sql_shuffle_partitions_conf(spark_opts: Dict[str, str]) -> Dict[str,
         str(spark_opts['spark.dynamicAllocation.maxExecutors']) != 'infinity'
     ):
 
-        num_partitions_dra = 2 * (
+        num_partitions_dra = 3 * (
             int(spark_opts.get('spark.dynamicAllocation.maxExecutors', 0)) *
```

Comment:
Maybe change this to `spark.dynamicAllocation.minExecutors`?

Reply:
How about? 
if dra_enabled:
   num_partitions = min_executor_instances * cores * 2
else: 
   num_partitions = executor_instances * cores * 2

Otherwise, if you can add comment something in lines of, "We are inclining in the direction of more parallelism of tasks across max available cores"